sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -851,7 +851,7 @@ private[hive] object HiveClientImpl {
       hiveTable.setFields(schema.asJava)
     }
     hiveTable.setPartCols(partCols.asJava)
-    conf.foreach(c => hiveTable.setOwner(c.getUser))
+    conf.foreach(c => hiveTable.setOwner(SessionState.get().getAuthenticator().getUserName()))
```

Comment:
We cannot use the following since it was introduced in 0.13.
```
conf.foreach(c => hiveTable.setOwner(SessionState.getUserFromAuthenticator()))
```